### DISCUSSION TITLE
Sample code for object creation via pyd-file

### DISCUSSION THREAD

**Discussion by Nesavi** [pyscripter/python4delphi#473] [Q&A] [answered]
I'd like to be able to create instance of a given class, and do this via a pyd-file. In the demos there seems to "only" samples from GUI-applications? Is there a sample where this is done at run time?

**Comment by pyscripter** [comment, marked as answer]
Please see the examples of created python extension modules (pyd extension) in the Modules subdirectory.

**Reply by Nesavi** [reply]
I think I may have not formulated myself very clearly. What I was looking for was sample code, that allows me to expose the constructtion of Delphi types in Python. I went through the demo more carefully, and I believe what I am looking for is Demo34, where instances of a type `TPoint` are created in a Python-scrip.

**Reply by pyscripter** [reply]
You may want to have a look at https://github.com/pyscripter/python4delphi/blob/master/Tests/WrapDelphiTest.pas which shows/tests different ways of exposing delphi objects, interfaces and records.
